record/record.go

```diff
@@ -169,7 +177,7 @@ func (r *Reader) nextChunk(wantFirst bool) error {
 			r.last = chunkType == fullChunkType || chunkType == lastChunkType
 			return nil
 		}
-		if r.n < blockSize && r.started {
+		if r.n < blockSize && r.started && !r.inRecovery {
 			if r.j != r.n {
 				return io.ErrUnexpectedEOF
 			}
```

Comment:
While you're here, please delete the panic at the bottom of this function. It should no longer be necessary, as of Go 1.1.